### PR TITLE
add debugger and task provider

### PR DESCRIPTION
**Pull request by midgethetree**
With these commits, users can use Run and Debug to run the current workspace in Ren'Py. This is now the default launch option for .rpy scripts when one is open in the active editor and there is no launch.json file setting a different default launch option. If the user chooses to generate a launch.json file for the current workspace, they can choose this debugger to get a launch.json file preconfigured to run that workspace in Ren'Py as the default launch option for the entire workspace.

These commits also provide four new tasks: lint, which lints the current workspace and saves it in lint.txt in the project folder; compile, which compiles the Ren'Py scripts in the current workspace; rmpersistent, which removes persistent data tied to the current workspace; and dialogue, which extracts dialogue from the current workspace in the default language.

Launch.json and tasks.json can be used to configure these new options and add additional launch options/tasks (to implement other renpy commands like translate, have multiple versions of the dialogue task using different arguments, have launch options with pretasks, etc.).

Finally, I fixed the existing run renpy command to utilize the new debugger instead of a "cmd" debugger users may not have an extension for. I also ensured the previously-used config option for setting the renpy executable's location actually appears as a setting in the vscode interface for editing, which is necessary for the debugger, tasks, and run command to work properly.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,7 +36,8 @@
   },
   "activationEvents": [
     "workspaceContains:**/*.rpy",
-    "workspaceContains:**/_ren.py"
+    "workspaceContains:**/_ren.py",
```
Comment (duckdoom4) [inline package.json:39]: Ah nice catch. I forgot to add this one :)

File: src/extension.ts [hunk 2]
```diff
@@ -248,12 +252,41 @@ export async function activate(context: ExtensionContext): Promise<void> {
...
+
+    renpyTaskProvider = tasks.registerTaskProvider("renpy", new RenpyTaskProvider());
+
     logMessage(LogLevel.Info, "Ren'Py extension activated!");
 }
 
 export function deactivate() {
     logMessage(LogLevel.Info, "Ren'Py extension deactivating");
     fs.unwatchFile(getNavigationJsonFilepath());
+    if (renpyTaskProvider) {
```
Comment (duckdoom4) [inline src/extension.ts:287]: You can use the extension subscription register to automatically discard this.
`context.subscriptions.push(`

But I see you did do that in other places, so maybe there's a reason you didn't do it here?

**Review by duckdoom4 — CHANGES_REQUESTED** [review]
Really awesome stuff! Thanks a lot for your contribution :)

Just one small question and then we can get this merged!

**Comment by duckdoom4** [comment]
One more thing, could you swap the target branch to develop? 
We push new features to develop until we are certain nothing is broken. Then we merge develop back into master when we think everything is ready.

File: src/extension.ts (hunk 2 above)
Comment (midgethetree) [reply src/extension.ts]: I was just copying that from the example task provider extension in the microsoft repo (https://github.com/microsoft/vscode-extension-samples/blob/main/task-provider-sample/src/extension.ts), but you're right that context.subscriptions should still work for this too, so I've changed that.

**Review by duckdoom4 — APPROVED** [review]
Awesome work, let get this merged! 🎉

**Pull request merged**